modules/internal/DefaultSparse.chpl

```diff
@@ -235,7 +235,114 @@ module DefaultSparse {
         add_help(ind);
       }
     }
-  
+
+    proc bulkAdd(inds: [] index(rank, idxType), sorted=false, noDuplicate=false){
+      var numAdded = inds.size; //maybe remove this var
+
+      //find individual insert points
+      //and eliminate duplicates between inds and dom
+      var indivInsertPts: [{0..#numAdded}] int;
+      var actualInsertPts: [{0..#numAdded}] int; //where to put in newdom
+
+      if !sorted then QuickSort(inds);
+
+      //eliminate duplicates --assumes sorted
+      if !noDuplicate {
+        //make sure lastInd != inds[inds.domain.low]
+        var lastInd = inds[inds.domain.low] + 1; 
+        for (i, p) in zip(inds, indivInsertPts)  {
+          if i == lastInd then p = -1;
```

Comment:
Use of -1 in a bunch of places here might be better handled by parentDom.low - 1 or something like that. -1 wouldn't work with unsigned index types. Of course, it wouldn't surprise me if there are other problems with unsigned index types for sparse domains... so it might be moot.